Engine/ResourceManagement/Importer/TextureImporter.cpp

```diff
@@ -106,65 +88,121 @@ ILubyte * TextureImporter::LoadImageDataInMemory(const std::string& file_path, i
 	{
-		return texture_file_path;
+		image_type = IL_BMP;
 	}
-	else
+	else if (file_extension == "cut")
 	{
-		std::string texture_filename = texture_file_path.substr(found, texture_file_path.length());
-
-		return texture_filename;
+		image_type = IL_CUT;
 	}
-}
-
-std::string TextureImporter::ImportToDDS(const File & file) const
-{
-	ILuint image;
-	ilGenImages(1, &image);
-	ilBindImage(image);
-	int width, height;
-	ILubyte * save_data = LoadImageDataInMemory(file.file_path.c_str(), IL_RGBA, width, height);
-	//Get new Name
-
-	std::string texture_name_no_extension = file.filename.substr(0, file.filename.find_last_of("."));
-	std::string output_file = SaveMetaFile(file.file_path, ResourceType::TEXTURE);
-
-	//Save data
-	ILuint size;
-
-	size = ilSaveL(IL_DDS, NULL, 0); // Get the size of the data buffer
-	if (size > 0) {
-
-		if (ilSaveL(IL_DDS, save_data, size) > 0) // Save to buffer with the ilSaveIL function
-		{
-			App->filesystem->Save(output_file.c_str(), (unsigned int *)save_data, size);
-		}
-		ilDeleteImages(1, &image);
+	else if (file_extension == "dds")
+	{
+		image_type = IL_DDS;
 	}
-	return output_file;
-}
-
-std::string TextureImporter::ImportToTGA(const File & file) const
-{
-	std::string output_file = SaveMetaFile(file.file_path, ResourceType::TEXTURE);
-	bool copied = App->filesystem->Copy(file.file_path.c_str(), output_file.c_str());
-	if (!copied)
+	else if (file_extension == "gif")
+	{
+		image_type = IL_GIF;
+	}
+	else if (file_extension == "ico")
+	{
+		image_type = IL_ICO;
+	}
+	else if (file_extension == "jpg")
+	{
+		image_type = IL_JPG;
+	}
+	else if (file_extension == "lif")
+	{
+		image_type = IL_LIF;
+	}
+	else if (file_extension == "mng")
+	{
+		image_type = IL_MNG;
+	}
+	else if (file_extension == "pcd")
+	{
+		image_type = IL_PCD;
+	}
+	else if (file_extension == "pcx")
+	{
+		image_type = IL_PCX;
+	}
+	else if (file_extension == "pic")
+	{
+		image_type = IL_PIC;
+	}
+	else if (file_extension == "png")
+	{
+		image_type = IL_PNG;
+	}
+	else if (file_extension == "pbm")
+	{
+		image_type = IL_PNM;
+	}
+	else if (file_extension == "pgm")
+	{
+		image_type = IL_PNM;
+	}
+	else if (file_extension == "ppm")
+	{
+		image_type = IL_PNM;
+	}
+	else if (file_extension == "psd")
+	{
+		image_type = IL_PSD;
+	}
+	else if (file_extension == "psp")
+	{
+		image_type = IL_PSP;
+	}
+	else if (file_extension == "bw")
+	{
+		image_type = IL_SGI;
+	}
+	else if (file_extension == "rgb")
+	{
+		image_type = IL_SGI;
+	}
+	else if (file_extension == "rgba")
+	{
+		image_type = IL_SGI;
+	}
+	else if (file_extension == "sgi")
 	{
-		return "";
+		image_type = IL_SGI;
 	}
-	return output_file;
+	else if (file_extension == "tga")
+	{
+		image_type = IL_TGA;
+	}
+	else if (file_extension == "tif")
+	{
+		image_type = IL_TIF;
+	}
+	else if (file_extension == "tiff")
+	{
+		image_type = IL_TIF;
+	}
+	else if (file_extension == "jasc")
+	{
+		image_type = IL_JASC_PAL;
+	}
+	else
+	{
+		image_type = IL_TYPE_UNKNOWN;
+	}
```

Comment:
maybe using a map instead those if-else statements?

Reply:
Here is the conflict of using execution time (current solution) vs using memory (map solution). Because this is done just when importing, I choose this solution instead of having memory allocated and unnused.